### PR TITLE
Improve performance of the state transitions panel on large messages

### PR DESCRIPTION
**Pull request by foxymiles**
**User-Facing Changes**
Improve performance of the state transitions panel on large messages

**Description**
Improve performance of the state transitions panel on large messages by taking advantage of our new sliced topic subscriptions.

### CODE REVIEW

File: packages/studio-base/src/PanelAPI/useBlocksSubscriptions.test.tsx [hunk 1]
```diff
@@ -16,13 +16,14 @@ import { renderHook } from "@testing-library/react-hooks";
 import { cloneDeep } from "lodash";
 
 import MockMessagePipelineProvider from "@foxglove/studio-base/components/MessagePipeline/MockMessagePipelineProvider";
+import { SubscribePayload } from "@foxglove/studio-base/players/types";
 
 import * as PanelAPI from ".";
 
-describe("useBlocksByTopic", () => {
+describe("useBlocksSubscriptions", () => {
```
Comment (snosenzo) [inline packages/studio-base/src/PanelAPI/useBlocksSubscriptions.test.tsx:23]: should we add some tests for `fields` subscriptions as well? Not seeing that represented here

File: packages/studio-base/src/panels/StateTransitions/index.tsx [hunk 2]
```diff
@@ -196,7 +196,20 @@ const StateTransitions = React.memo(function StateTransitions(props: Props) {
 
   const decodeMessagePathsForMessagesByTopic = useDecodeMessagePathsForMessagesByTopic(pathStrings);
 
-  const blocks = useBlocksByTopic(subscribeTopics);
+  const subscriptions: SubscribePayload[] = useMemo(
+    () =>
+      filterMap(pathStrings, (path) => {
+        const payload = subscribePayloadFromMessagePath(path, "full");
+        // Include the header in case we are ordering by header stamp.
+        if (payload?.fields != undefined) {
```
Comment (snosenzo) [inline packages/studio-base/src/panels/StateTransitions/index.tsx:204]: In the plot panel do we always request header as well? or only when ordering by header stamp is active?

**Review by snosenzo — APPROVED** [review]
couple questions/consideration, otherwise looks good

File: packages/studio-base/src/PanelAPI/useBlocksSubscriptions.test.tsx (hunk 1 above)
Comment (foxymiles) [reply packages/studio-base/src/PanelAPI/useBlocksSubscriptions.test.tsx:23]: Yes good idea. I'll add some.

File: packages/studio-base/src/panels/StateTransitions/index.tsx (hunk 2 above)
Comment (foxymiles) [reply packages/studio-base/src/panels/StateTransitions/index.tsx:204]: In plot we're a bit smarter and only add it if header stamp is requested. I'll rearrange this a bit to use the same logic.

**Pull request merged**